src/Unleash/DefaultUnleash.cs

```diff
@@ -286,6 +287,7 @@ private void EmitImpressionEvent(string type, UnleashContext context, bool enabl
             catch (Exception ex)
             {
                 Logger.Error($"UNLEASH: Emitting impression event callback threw exception: {ex.Message}");
+                EventConfig?.RaiseError(new ErrorEvent() { Error = ex, ErrorType = ErrorType.ImpressionEvent });
```

Comment:
This feels brave. What situations can cause the impression events to fail that wouldn't also impact the error events?

Reply:
This is one of the things I was a bit uncertain of. But it can happen if I implement the impression event callback and then my code throws. Maybe not a good fit for the ErrorEvent as technically it's source is external to our code